### PR TITLE
Split resource variables

### PR DESCRIPTION
**Pull request by Twsouza**
Split the `${RESOURCE_NAME}_URL` into : 

```
${RESOURCE_NAME}_USER
${RESOURCE_NAME}_PASS
${RESOURCE_NAME}_HOST
${RESOURCE_NAME}_PORT
${RESOURCE_NAME}_NAME
```

### CODE REVIEW

File: provider/k8s/process.go [hunk 1]
```diff
@@ -428,12 +428,14 @@ func (p *Provider) podSpecFromService(app, service, release string) (*ac.PodSpec
 			c.Image = fmt.Sprintf("%s:%s.%s", repo, service, r.Build)
 
 			for _, r := range s.ResourceMap() {
+				key := strings.Join(strings.Split(r.Env, "_")[1:], "_")
+				fmt.Printf("\n\n\n========> key for %v: %v\n\n\n", r.Env, key)
```
Comment (nightfury1204) [inline provider/k8s/process.go:432]: is this print line necessary?

**Review by yuriadams — APPROVED** [review]

**Review by nightfury1204 — APPROVED** [review]
:fire:

File: provider/k8s/process.go [hunk 2]
```diff
@@ -428,12 +428,13 @@ func (p *Provider) podSpecFromService(app, service, release string) (*ac.PodSpec
 			c.Image = fmt.Sprintf("%s:%s.%s", repo, service, r.Build)
 
 			for _, r := range s.ResourceMap() {
+				key := strings.Join(strings.Split(r.Env, "_")[1:], "_")
```
Comment (beastawakens) [inline provider/k8s/process.go:431]: @Twsouza This breaks things for any resource with a 'multi-part' name.
i.e. I have a resource called `primary-database` in the convox.yml.  This means I end up with a env var called `PRIMARY_DATABASE_URL`.  
But splitting on the underscore and pulling out everything after the first part, means that the `key` is now `DATABASE_URL` which doesn't exist in the ConfigMap so deployments fail as that can't be found.

File: provider/k8s/template.go [hunk 3]
```diff
@@ -82,6 +82,9 @@ func (p *Provider) templateHelpers() template.FuncMap {
 		"k8sname": func(s string) string {
 			return nameFilter(s)
 		},
+		"resourceEnv": func(s string) string {
+			return strings.Join(strings.Split(s, "_")[1:], "_")
```
Comment (beastawakens) [inline provider/k8s/template.go:86]: I think this will need to be changed as well.

File: provider/k8s/process.go (hunk 2 above)
Comment (Twsouza) [reply provider/k8s/process.go:431]: hey @beastawakens thanks for pointing this out. I've opened a PR to fix it, see https://github.com/convox/convox/pull/517

**Pull request merged**